### PR TITLE
Added 2 null checks for new game challenge stage and p4g

### PR DESCRIPTION
**Pull request by EXtremeExploit**
Further into gameplay in new game challenge stage and p4g
https://github.com/Vita3K/compatibility/issues/107
https://github.com/Vita3K/compatibility/issues/155

### CODE REVIEW

**Comment by Zangetsu38** [comment]
name commit is wrong

**Comment by 1whatleytay** [comment]
No need to worry about the commit names for now. I can squash.

**Comment by Zangetsu38** [comment]
ah yes forget ^^

File: vita3k/modules/SceGxm/SceGxm.cpp [hunk 1]
```diff
@@ -1035,6 +1035,8 @@ EXPORT(Ptr<const char>, sceGxmProgramParameterGetName, Ptr<const SceGxmProgramPa
 }
 
 EXPORT(unsigned int, sceGxmProgramParameterGetResourceIndex, const SceGxmProgramParameter *parameter) {
+    if (parameter == nullptr)
+        return {};
```
Comment (1whatleytay) [inline vita3k/modules/SceGxm/SceGxm.cpp:1039]: Do something like `RET_ERROR(0)` maybe?

File: vita3k/modules/SceGxm/SceGxm.cpp [hunk 2]
```diff
@@ -1272,6 +1274,9 @@ EXPORT(int, sceGxmSetUniformDataF, void *uniformBuffer, const SceGxmProgramParam
     assert(componentCount > 0);
     assert(sourceData != nullptr);
 
+    if (uniformBuffer == nullptr)
+        return 0;
```
Comment (1whatleytay) [inline vita3k/modules/SceGxm/SceGxm.cpp:1278]: I would go with `RET_ERROR(SCE_GXM_ERROR_INVALID_PARAMETER)` or something.

**Comment by 1whatleytay** [comment]
No need to add a new error.

**Pull request merged**